src/Credentials/AssumeRoleWithWebIdentityCredentialProvider.php

```diff
@@ -94,12 +94,13 @@ public function __invoke()
             $result = null;
             while ($result == null) {
                 try {
-                    $token = file_get_contents($this->tokenFile);
-                    if (false === $token) {
+                    if (is_readable($this->tokenFile)) {
```

Comment:
Is the main purpose to throw a `CredentialsException` instead of having the file warning? If so, if the `is_readable` call fails, won't it reach the same warning within the `else` block?